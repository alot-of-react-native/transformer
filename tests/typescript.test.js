import fs from 'fs';
import path from 'path';

process.env.PLATFORM_TARGET = 'tests';
const transformer = require('../dist');
const FIXTURES = '__fixtures__';

const GOOD_FIXTURES_DIR = path.join(__dirname, FIXTURES, 'good');
const BAD_FIXTURES_DIR = path.join(__dirname, FIXTURES, 'bad');

describe('@alot/transformer', () => {
  fs.readdirSync(GOOD_FIXTURES_DIR).forEach(fixture => {
    const filename = path.join(GOOD_FIXTURES_DIR, fixture);
    if (fs.statSync(filename).isFile()) {
      it(`transpiles ${filename}`, () => {
        const src = fs.readFileSync(filename, { encoding: 'utf8' });
        const result = transformer.transform({src, filename, options: {}});
        expect(result.code).toMatchSnapshot();
        expect(result.map).toMatchSnapshot();
      });
    }
  });
});
