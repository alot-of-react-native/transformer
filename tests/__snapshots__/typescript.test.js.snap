// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@alot/transformer transpiles P:\\c\\gl\\react-native-alot-transformer\\tests\\__fixtures__\\good\\leftPad.ts 1`] = `
"\\"use strict\\";

var cache = ['', ' ', '  ', '   ', '    ', '     ', '      ', '       ', '        ', '         '];
function leftPad(str, len, ch) {
    if (ch === void 0) {
        ch = ' ';
    }

    str = str + '';

    len = len - str.length;

    if (len <= 0) return str;

    if (!ch && ch !== 0) ch = ' ';

    ch = ch + '';

    if (ch === ' ' && len < 10) return cache[len] + str;

    var pad = '';

    while (true) {
        if (len & 1) pad += ch;

        len >>= 1;

        if (len) ch += ch;else break;
    }

    return pad + str;
}"
`;

exports[`@alot/transformer transpiles P:\\c\\gl\\react-native-alot-transformer\\tests\\__fixtures__\\good\\leftPad.ts 2`] = `
Array [
  Array [
    3,
    0,
    1,
    0,
  ],
  Array [
    3,
    4,
    1,
    6,
    "cache",
  ],
  Array [
    3,
    12,
    1,
    24,
  ],
  Array [
    3,
    13,
    2,
    2,
  ],
  Array [
    3,
    15,
    1,
    24,
  ],
  Array [
    3,
    17,
    3,
    2,
  ],
  Array [
    3,
    20,
    1,
    24,
  ],
  Array [
    3,
    22,
    4,
    2,
  ],
  Array [
    3,
    26,
    1,
    24,
  ],
  Array [
    3,
    28,
    5,
    2,
  ],
  Array [
    3,
    33,
    1,
    24,
  ],
  Array [
    3,
    35,
    6,
    2,
  ],
  Array [
    3,
    41,
    1,
    24,
  ],
  Array [
    3,
    43,
    7,
    2,
  ],
  Array [
    3,
    50,
    1,
    24,
  ],
  Array [
    3,
    52,
    8,
    2,
  ],
  Array [
    3,
    60,
    1,
    24,
  ],
  Array [
    3,
    62,
    9,
    2,
  ],
  Array [
    3,
    71,
    1,
    24,
  ],
  Array [
    3,
    73,
    10,
    2,
  ],
  Array [
    3,
    83,
    1,
    24,
  ],
  Array [
    3,
    85,
    11,
    2,
  ],
  Array [
    3,
    96,
    1,
    24,
  ],
  Array [
    3,
    97,
    1,
    0,
  ],
  Array [
    4,
    0,
    14,
    0,
  ],
  Array [
    4,
    9,
    14,
    0,
    "leftPad",
  ],
  Array [
    4,
    16,
    14,
    0,
  ],
  Array [
    4,
    17,
    14,
    17,
    "str",
  ],
  Array [
    4,
    20,
    14,
    0,
  ],
  Array [
    4,
    22,
    14,
    30,
    "len",
  ],
  Array [
    4,
    25,
    14,
    0,
  ],
  Array [
    4,
    27,
    14,
    43,
    "ch",
  ],
  Array [
    4,
    29,
    14,
    0,
  ],
  Array [
    4,
    31,
    14,
    68,
  ],
  Array [
    5,
    0,
    14,
    43,
  ],
  Array [
    5,
    8,
    14,
    43,
    "ch",
  ],
  Array [
    5,
    15,
    14,
    43,
  ],
  Array [
    5,
    20,
    14,
    43,
  ],
  Array [
    5,
    21,
    14,
    43,
  ],
  Array [
    5,
    23,
    14,
    43,
  ],
  Array [
    6,
    0,
    14,
    43,
    "ch",
  ],
  Array [
    6,
    13,
    14,
    43,
  ],
  Array [
    6,
    16,
    14,
    43,
  ],
  Array [
    7,
    0,
    14,
    68,
  ],
  Array [
    9,
    0,
    16,
    2,
    "str",
  ],
  Array [
    9,
    10,
    16,
    8,
    "str",
  ],
  Array [
    9,
    16,
    16,
    14,
  ],
  Array [
    9,
    18,
    16,
    2,
  ],
  Array [
    11,
    0,
    18,
    2,
    "len",
  ],
  Array [
    11,
    10,
    18,
    8,
    "len",
  ],
  Array [
    11,
    16,
    18,
    14,
    "str",
  ],
  Array [
    11,
    20,
    18,
    18,
    "length",
  ],
  Array [
    11,
    26,
    18,
    2,
  ],
  Array [
    13,
    0,
    20,
    2,
  ],
  Array [
    13,
    8,
    20,
    6,
    "len",
  ],
  Array [
    13,
    15,
    20,
    13,
  ],
  Array [
    13,
    16,
    20,
    2,
  ],
  Array [
    13,
    18,
    20,
    16,
  ],
  Array [
    13,
    25,
    20,
    23,
    "str",
  ],
  Array [
    13,
    28,
    20,
    16,
  ],
  Array [
    15,
    0,
    22,
    2,
  ],
  Array [
    15,
    8,
    22,
    6,
  ],
  Array [
    15,
    9,
    22,
    7,
    "ch",
  ],
  Array [
    15,
    11,
    22,
    6,
  ],
  Array [
    15,
    15,
    22,
    13,
    "ch",
  ],
  Array [
    15,
    22,
    22,
    20,
  ],
  Array [
    15,
    23,
    22,
    2,
  ],
  Array [
    15,
    25,
    22,
    23,
    "ch",
  ],
  Array [
    15,
    30,
    22,
    28,
  ],
  Array [
    15,
    33,
    22,
    23,
  ],
  Array [
    17,
    0,
    24,
    2,
    "ch",
  ],
  Array [
    17,
    9,
    24,
    7,
    "ch",
  ],
  Array [
    17,
    14,
    24,
    12,
  ],
  Array [
    17,
    16,
    24,
    2,
  ],
  Array [
    19,
    0,
    26,
    2,
  ],
  Array [
    19,
    8,
    26,
    6,
    "ch",
  ],
  Array [
    19,
    15,
    26,
    13,
  ],
  Array [
    19,
    18,
    26,
    6,
  ],
  Array [
    19,
    22,
    26,
    20,
    "len",
  ],
  Array [
    19,
    28,
    26,
    26,
  ],
  Array [
    19,
    30,
    26,
    2,
  ],
  Array [
    19,
    32,
    26,
    30,
  ],
  Array [
    19,
    39,
    26,
    37,
    "cache",
  ],
  Array [
    19,
    45,
    26,
    43,
    "len",
  ],
  Array [
    19,
    48,
    26,
    37,
  ],
  Array [
    19,
    52,
    26,
    50,
    "str",
  ],
  Array [
    19,
    55,
    26,
    30,
  ],
  Array [
    21,
    0,
    28,
    2,
  ],
  Array [
    21,
    8,
    28,
    6,
    "pad",
  ],
  Array [
    21,
    14,
    28,
    12,
  ],
  Array [
    21,
    16,
    28,
    2,
  ],
  Array [
    23,
    0,
    30,
    2,
  ],
  Array [
    23,
    11,
    30,
    9,
  ],
  Array [
    23,
    15,
    30,
    2,
  ],
  Array [
    23,
    17,
    30,
    15,
  ],
  Array [
    24,
    0,
    32,
    4,
  ],
  Array [
    24,
    12,
    32,
    8,
    "len",
  ],
  Array [
    24,
    18,
    32,
    14,
  ],
  Array [
    24,
    19,
    32,
    4,
  ],
  Array [
    24,
    21,
    32,
    17,
    "pad",
  ],
  Array [
    24,
    28,
    32,
    24,
    "ch",
  ],
  Array [
    24,
    30,
    32,
    17,
  ],
  Array [
    26,
    0,
    34,
    4,
    "len",
  ],
  Array [
    26,
    16,
    34,
    12,
  ],
  Array [
    26,
    17,
    34,
    4,
  ],
  Array [
    28,
    0,
    38,
    4,
  ],
  Array [
    28,
    12,
    38,
    8,
    "len",
  ],
  Array [
    28,
    15,
    38,
    4,
  ],
  Array [
    28,
    17,
    38,
    13,
    "ch",
  ],
  Array [
    28,
    23,
    38,
    19,
    "ch",
  ],
  Array [
    28,
    25,
    38,
    13,
  ],
  Array [
    28,
    26,
    38,
    4,
  ],
  Array [
    28,
    31,
    40,
    9,
  ],
  Array [
    29,
    0,
    41,
    3,
  ],
  Array [
    31,
    0,
    43,
    2,
  ],
  Array [
    31,
    11,
    43,
    9,
    "pad",
  ],
  Array [
    31,
    17,
    43,
    15,
    "str",
  ],
  Array [
    31,
    20,
    43,
    2,
  ],
  Array [
    32,
    0,
    44,
    1,
  ],
]
`;
